server/src/main/java/org/jboss/as/server/deployment/module/descriptor/JBossDeploymentStructureParser10.java

```diff
@@ -522,8 +522,22 @@ private static void parseResourceRoot(final DeploymentUnit deploymentUnit, final
                             Closeable closable = null;
                             if(overlay != null) {
                                 overlay.remountAsZip(false);
-                            } else if(child.isFile()) {
-                                closable = VFS.mountZip(child, child, TempFileProviderService.provider());
+                            } else {
+                                if(!child.exists()) {
+                                    // the reference does not exist
+                                    throw ServerLogger.ROOT_LOGGER.illegalResoureRootReference(child.getName());
+                                } else if(child.isFile()) {
+                                    try {
+                                        // check if the resource-root is within the deployment or not
+                                       if(child.getPathNameRelativeTo(deploymentRootFile) != null) {
```

Comment:
Let me rephrase that. What I mean is that `null` will never be returned. It doesn't seem quite right to use the exception thrown as the determining factor of whether it's outside the deployment or not.